### PR TITLE
Leadworks guns rework

### PR DESCRIPTION
**Pull request by Firestorm01X2**
Currently Leadworks guns mention that they have a lot of built in attachments. But in fact they contains none of that. Leadwork weapon stats pretend to reflect atthments effect. But in most cases missed effects from attachments.

For example L523-DSR rifle has bipods by its description but in fact it does not contains it so L523-DSR does not get stabilization bonus intended for bipods.
L523-MBR rifle in fact does not contain scope that mention in its description so it does not get reduced sight dispersion. And so on.

This PR going to change that.

Now all Leadworks guns have special irremovable proprietary(!) Leadworks attachments. 
Added a lot of irremovable proprietary(!) Leadworks attachments. Scopes, sights, barrels, gyro stabilization system , e.t.c
Most of them releated on existing ones with certain modification of stats.

So now all Leadworks gun actually will be Christmas Tree Guns and will have all necessary effect of attachments.

Changed some exisitng guns to make them more like Leadworks guns. 
For example L2037 Backup now come with built-in Holo sight and laser sight and does not accept addtitonal attachments.
# List of changes

## Idea
I want to make Leadworks "modular" manufacturer who pack his guns with attachments but do not allow custom modifications his guns and has hard warranty rules. But this way user get everything he (probably) need right of the box.

Not all Leadworks guns are modular but rules "No third party attachments" and "No custom modifications" are strict.

Leadworks milspec guns are mostly modular (L523 platform) but their modules also contains attachments that also can't be removed. If solder critically break some attachment - he can just replace whole module.

The only proper way to remove Leadworks attachments is using complex factory machnines and tools so it is very unlikely for player to be able to do it in current apocalyptic situation. 

However Leadworks guns designed to allow field repair and maintenance. To certain degree of course.

It sounds like bad idead at first, but getting gun, packed with everything you need "right of the box" is actually nice.

IRL there is XM29_OICW  https://ru.wikipedia.org/wiki/XM29_OICW . 
As far as know It didn't succeed commercially. And didn't succeed at all.

##  Minor changes
* **Rail laser sight sight diespersion set to 30 from 360**. I forgort to change during ranged rebalance. It is not woth separate PR so I did it here.
* **Fixed typo in M320 GLM dispersion**. It should be 270 not 20 (non module version has dispersion 270). I've derived from it so I have fo fix its dispersion.

##  Rules
* **All Leadwork guns can't install attachment now except of accesories. Now it applies to L523 modular system and separate guns including guns for civilian use. Instead guns already comes with wide array of attachments. All Leadworks attachments are irremovable.** 
* All Leadworks attachments are irremovable have no weight and volume. Weight and volume set in weapon istelf since attahcments are part of module.
* Pisotls (and shotgun) come with red dot and laser sight.
* Revolvers come with holo sight and laser sight.

## New attachments

* Leadworks red dot sight - copy of normal red dot.
* Leadworks sniper scope - copy of rifle scope.
* Leadworks ACOG scope - copy of ACOG scope.
* Leadworks holographic sight - copy of holo sight.
* **Leadworks barrel extension** - increase accuracy less then usual long barrel but at the same time increase weapon range and damage.
* **Leadworks shortened barrel** - does not deacrease accurasy as usual shortened barrel but decrease max wepon range.
* **Leadworks heavy duty barrel** - increase weapon damage and range. Does not increase accuracy. Designed for LMG.
* Leadworks gyroscopic stabilizer - same as stabilizer.
* Leadworks underbarrel laser sight - same as underbarrel laser sight.
* Leadworks forward grip - same as grip
* **Leadworks bipod** - same as bipod.
* **Leadworks underslung shotgun** -Copy of underslung shotgun. Used in carbine.
* Leadworks GLM- copy of M320 GLM.

## L523 modular system (Leadworks milspec rifles)

* L523 platform - dispersion increased to 140 from 90. Now there are a lof of mods that will decrease it in exact guns. Also removed iron sight to make naked platform harder to use.
* L523-CAR carbine - comes with Leadworks shortened barrel, Leadworks red dot sight, Leadworks underbarrel laser sight, satbilizer , and **Leadwoks underslung shotgun*** . New thing here is underslung shotgun.
* L523-DSR rifle - comes with Sniper scope, **Leadworks barrel extension** , bipod, grip and stabilizer.
* L523-LMG - comew with stabilizer, grip, **Leadworks heavy duty barrel**. **Now it comes with ACOG and bipod.** Added +45 dipersion to reflect that it is LMG (LMGs have more dispersion then rifles ingame).
* L523-MBR rifle - comes with acog scope and red-dot  sight, grip. **Added stabilizer and grenade luncher module.** Why stabilizer was not there in the first place since even carbine has that? GLM added to make MBR more distinct from other Leadworks rilfe platforms.

## Rifles
* L1820 Long Ranger -  comes with red dot and laser sight. No extra attachments allowed.

## Shotguns
* L12 Defender - comes with red dot and laser sight. Removed ablity to add attachments.

## Pistols
* L39  - comes with red dot and laser sight. No extra attachments allowed.
* L39-45 - comes with red dot and laser sight. No extra attachments allowed.
* L39B  - comes with red dot and laser sight. No extra attachments allowed.
* L39B-45 - comes with red dot and laser sight. No extra attachments allowed.
* L2031 Enforcer - comes with holo sight and laser sight. No extra attachments allowed.
* L2032 Lookout - come with holo sight and laser sight. No extra attachments allowed.
* L2037 Backup - come with holo sight and laser sight. No extra attachments allowed.

### CODE REVIEW

File: data/json/items/gunmod/barrel.json [hunk 1]
```diff
@@ -72,5 +72,39 @@
...
+  },
+  {
+    "id": "lead_barrel_heavy_duty",
+    "copy-from": "barrel_big",
+    "type": "GUNMOD",
+    "name": "Leadworks heavy duty barrel",
+    "description": "A Leadworks built in heavy duty barrel especially designed for prolonged shooting. Reduce accuracy but increase damage output and weapon range.",
+    "price": 0,
+    "damage_modifier": 3,
+    "dispersion_modifier": 60,
```
Comment (ymber) [inline data/json/items/gunmod/barrel.json]: What's the rationale for these numbers? How does it increase muzzle velocity and effective range while reducing accuracy?

Comment (Firestorm01X2) [reply data/json/items/gunmod/barrel.json]: It is debuff for LMG. Originally L523-LMG has +60 dispersion. Now it comes from that mod.

Comment (Firestorm01X2) [reply data/json/items/gunmod/barrel.json]: In fact it should be 90 bacues gyro stybilizer reduce dispersion to 30.

File: data/json/items/gun/223.json [hunk 2]
```diff
@@ -153,17 +154,15 @@
...
     "modes": [
       [ "DEFAULT", "semi-auto", 1 ],
       [ "AUTO", "auto", 5 ]
     ],
-    "relative": {
-      "weight": 400,
-      "range": -6
-    }
+    "built_in_mods": [ "lead_red_dot_sight", "lead_barrel_small", "lead_laser_sight", "lead_stabilizer" ],
+	"valid_mod_locations": [ [ "accessories", 2 ], [ "sights", 1 ], [ "barrel", 1 ], [ "underbarrel", 1 ], [ "rail", 1 ] ]
```
Comment (Night-Pryanik) [inline data/json/items/gun/223.json]: Tab.

File: data/json/items/gun/223.json [hunk 3]
```diff
@@ -107,7 +107,7 @@
     "name": "L2037 Backup",
     "name_plural": "L2037 Backups",
     "//": "Most real-life autorevolvers use 6-7.5 inch barrel, but this one's mentioned to be particularly massive.",
-    "description": "Developed by Leadworks as a companion sidearm to the L523 modular system, this massive autorevolver fires the powerful .223 round, but suffers high recoil as a result.  It has a weighted barrel to make recoil more manageable and a 5-chamber cylinder.",
+    "description": "Developed by Leadworks as a companion sidearm to the L523 modular system, this massive autorevolver fires the powerful .223 round, but suffers high recoil as a result. It has a weighted barrel to make recoil more manageable and a 5-chamber cylinder. Holographics and laser sights built in for additonal accuracy. Similar to other Leadworks products it doesn't accept third-party modifications.",
```
Comment (Night-Pryanik) [inline data/json/items/gun/223.json]: We have a convention that there should be two spaces after periods.

File: data/json/items/gun/223.json [hunk 4]
```diff
@@ -172,53 +171,41 @@
...
+    "built_in_mods": [ "lead_rifle_scope", "lead_barrel_big", "lead_bipod", "lead_grip",  "lead_stabilizer" ],
+    "valid_mod_locations": [ [ "accessories", 2 ], [ "sights", 1 ], [ "barrel", 1 ] , [ "underbarrel", 2 ], [ "rail", 1 ] ]
   },
   {
     "id": "l_lmg_223",
     "copy-from": "l_base_223",
     "type": "GUN",
     "name": "L523-LMG",
-    "description": "Leadworks LLC's bullpup L523 platform utilizes the ubiquitous 5.56 round in multiple applications, and is designed to part-swap quickly, safely, and effectively from one to the next.  The rather ambitious light machine gun variant incorporates a heavy-duty barrel & foregrip as well as the Leadworks gyro system.  Unlike the other configurations, the L523 LMG can be loaded with ammo belts.",
+    "description": "Leadworks LLC's bullpup L523 platform utilizes the ubiquitous 5.56 round in multiple applications, and is designed to part-swap quickly, safely, and effectively from one to the next.  The rather ambitious light machine gun variant incorporates ACOG scope, a heavy-duty barrel, foregrip, bipods and as well as the Leadworks gyro system.  Unlike the other configurations, the L523 LMG can be loaded with ammo belts.",
```
Comment (Night-Pryanik) [inline data/json/items/gun/223.json]: Bipod, not bipods.

File: data/json/items/gunmod/barrel.json [hunk 5]
```diff
@@ -72,5 +72,39 @@
     "dispersion_modifier": 500,
     "ups_charges": 50,
     "min_skills": [ [ "gun", 1 ] ]
+  },
+  {
+    "id": "lead_barrel_big",
+    "copy-from": "barrel_big",
+    "type": "GUNMOD",
+    "name": "Leadworks barrel extension",
+    "description": "A Leadworks built in longer barrel increases the muzzle velocity of a firearm, contributing to both accuracy and damage slows aiming.",
```
Comment (Night-Pryanik) [inline data/json/items/gunmod/barrel.json]: "..._but_ slows aiming"?

File: data/json/items/gunmod/barrel.json [hunk 6]
```diff
@@ -72,5 +72,39 @@
...
+    "dispersion_modifier": 60,
+    "range": -6,
+    "flags": [ "IRREMOVABLE" ]
+  },
+  {
+    "id": "lead_barrel_heavy_duty",
+    "copy-from": "barrel_big",
+    "type": "GUNMOD",
+    "name": "Leadworks heavy duty barrel",
+    "description": "A Leadworks built in heavy duty barrel especially designed for prolonged shooting. Reduce accuracy but increase damage output and weapon range.",
```
Comment (Night-Pryanik) [inline data/json/items/gunmod/barrel.json]: Two spaces after period.

File: data/json/items/gunmod/sights.json [hunk 7]
```diff
@@ -178,5 +178,51 @@
...
+    "description": "An Leadworks built in Advanced Combat Optical Gunsight, a milspec medium magnification scope.  Increases weight but improves accuracy significantly.",
+    "price": 0,
+    "flags": [ "DISABLE_SIGHTS", "ZOOM", "IRREMOVABLE" ]
+  },
+  {
+    "id": "lead_acog_red_dot_scope",
+    "copy-from": "lead_acog_scope",
+    "type": "GUNMOD",
+    "name": "Leadworks ACOG and red dot combo scope",
+    "description": "An Leadworks built in (proprietary) combination optic & red-dot sights for quick and accurate targeting.",
```
Comment (Night-Pryanik) [inline data/json/items/gunmod/sights.json]: "... combination _of_ optic ..."?

File: data/json/items/gunmod/underbarrel.json [hunk 8]
```diff
@@ -324,5 +324,32 @@
...
+    "description": "A Leadworks built in grip placed forward on the barrel allows for greater control.",
+    "price": 0,
+    "flags": [ "PUMP_RAIL_COMPATIBLE", "IRREMOVABLE" ]
+  },
+  {
+    "id": "lead_bipod",
+    "copy-from": "bipod",
+    "type": "GUNMOD",
+    "name": "Leadworks bipod",
+    "description": "Leadworks built in bipods designed to provide a forward rest and reduce motion.  Although they greatly improve handling of recoil they are usable only on certain surfaces and are slow to equip.",
```
Comment (Night-Pryanik) [inline data/json/items/gunmod/underbarrel.json]: Bipod, not bipods. Also, singular, so "it", not "they".

File: data/json/items/gunmod/barrel.json (hunk 1 above)
Comment (ymber) [reply data/json/items/gunmod/barrel.json]: +90 is even worse. Increasing muzzle velocity and effective range while decreasing accuracy has no grounding in reality. If the weapon's stats were originally way out they need fixing here.

**Comment by Firestorm01X2** [comment]
Note that I write "Added" I mean that it was not in the despription. Before nono oe Leadworks gun contains any attahments

**Comment by Firestorm01X2** [comment]
Ready.

File: data/json/items/gun/45.json [hunk 9]
```diff
@@ -58,7 +58,7 @@
     "name": "L2031 Enforcer",
     "name_plural": "L2031 Enforcer",
     "//": "Real-life Autorevolvers tended to vary between 6-7.5 inch barrels, so we'll err on the side of a +1 here.",
-    "description": "An \"autorevolver\", this Leadworks LLC six-shooter combines the ease-of-action of a semi-auto with the ease-of-reloading and style of a revolver.  It achieved overnight stardom when it appeared in \"Treasure Trail\", a popular space-Western series running from 2030 through 2037.",
+    "description": "An \"autorevolver\", this Leadworks LLC six-shooter combines the ease-of-action of a semi-auto with the ease-of-reloading and style of a revolver.  It achieved overnight stardom when it appeared in \"Treasure Trail\", a popular space-Western series running from 2030 through 2037.  Come with built in holographic and laser sights.  Similar to other Leadworks products it doesn't accept third-party modifications.",
```
Comment (Rivet-the-Zombie) [inline data/json/items/gun/45.json]: >Come with

Should be:

>Comes with

File: data/json/items/gun/45.json [hunk 10]
```diff
@@ -113,7 +115,7 @@
     "copy-from": "l_sp_9mm",
     "type": "GUN",
     "name": "L39-45",
-    "description": "Capitalizing on the L39B-45's reputation as one of the most intimidating pistols on the market, Leadworks LLC manufactured a semiautomatic model as well.  Locking it semi-auto reduced cost and improved performance.  Similar to other Leadworks products it doesn't accept third-party modifications.",
+    "description": "Capitalizing on the L39B-45's reputation as one of the most intimidating pistols on the market, Leadworks LLC manufactured a semiautomatic model as well.  Locking it semi-auto reduced cost and improved performance.  Come with built in red dot and laser sights.  Similar to other Leadworks products it doesn't accept third-party modifications.",
```
Comment (Rivet-the-Zombie) [inline data/json/items/gun/45.json]: Same.

File: data/json/items/gun/9mm.json [hunk 11]
```diff
@@ -207,7 +207,7 @@
     "type": "GUN",
     "name": "L39B",
     "//": "The M93R can go through up to 110 rounds in six seconds, FWIW.",
-    "description": "A selective-firing 9x19mm handgun introduced in 2039, Leadworks LLC's L39B provides accurate single-shot placement as well as controllable short-range firepower with its three-round burst mode.  Users praised its accuracy and fearsome appearance.  Similar to other Leadworks products it doesn't accept third-party modifications.",
+    "description": "A selective-firing 9x19mm handgun introduced in 2039, Leadworks LLC's L39B provides accurate single-shot placement as well as controllable short-range firepower with its three-round burst mode.  Users praised its accuracy and fearsome appearance.  Come with built in red dot and laser sights.  Similar to other Leadworks products it doesn't accept third-party modifications.",
```
Comment (Rivet-the-Zombie) [inline data/json/items/gun/9mm.json]: Same.

**Comment by Firestorm01X2** [comment]
I need to make final adjustment. So WIP.

File: data/json/items/gun/9mm.json [hunk 12]
```diff
@@ -197,7 +197,7 @@
     "type": "GUN",
     "name": "L2032 Lookout",
     "name_plural": "L2032 Lookout",
-    "description": "After the Enforcer's success, Leadworks LLC reworked their autorevolver design to cater to the 9x19mm market.  The Lookout, released in 2032, takes advantage of the smaller 9x19mm to hold a seventh round, without sacrificing the ease of handling or reloading the Enforcer made legendary.",
+    "description": "After the Enforcer's success, Leadworks LLC reworked their autorevolver design to cater to the 9x19mm market.  The Lookout, released in 2032, takes advantage of the smaller 9x19mm to hold a seventh round, without sacrificing the ease of handling or reloading the Enforcer made legendary.  Comes with built in holographic and laser sights.  Similar to other Leadworks products it doesn't accept third-party modifications.",
```
Comment (Night-Pryanik) [inline data/json/items/gun/9mm.json]: I suppose we need to change the dates (or remove them altogether) here and below for the same reason it was done in https://github.com/CleverRaven/Cataclysm-DDA/commit/77b874ca4ce878ce4d9fe0c7a60ac1dd48cefe6e#diff-f66af8527da2e94290d5777a157b734a.

**Comment by Mecares** [comment]
Just wondering: what is the balancing you are aiming for? Are the Leadworks weapons supposed to be better than guns without mods but slightly worse than guns with mods or are they supposed to be on par with rare guns with mods or even superior?

**Comment by Firestorm01X2** [comment]
>1) Are the Leadworks weapons supposed to be better than guns without mods but slightly worse than guns with mods
>2) supposed to be on par with rare guns with mods or even superior?

Between 1 and 2. Closer to 1.

**Comment by Firestorm01X2** [comment]
@Mecares 
You know, you probably make a point there. I've reduced dispersion of L523 platform to 150.

**Comment by Firestorm01X2** [comment]
Ok. I should stop now.
Ready.

**Comment by Firestorm01X2** [comment]
Also. I've used underbarrel version of laser sight and grip for Leadwork attacments instead of rail and grip slot versions. Usual rifle has only one underbarrel slot. Leadworks got 2.
Anyway - it should not make a difference.

Ready.

**Comment by Firestorm01X2** [comment]
I need to think about carbine. Who the hell in the right mind will integrate bayonet in weapon hull in 21 centuary? I'll replace bayonet with something else.

**Comment by Firestorm01X2** [comment]
I've added built in underslung shotgun to L523-CAR carbine. It suits it beter thhen 20x ceturary pointy thing.

**Comment by Firestorm01X2** [comment]
Ready.

**Comment by Firestorm01X2** [comment]
Replaced combo scope (hybrid acog+red dot) on L523-MBR rifle to  2 separate sight: acog and red-dot. Game can handle multiple scopes properly. Let it do it. Separate entity for combo scope not worth it and was removed.

**Comment by Firestorm01X2** [comment]
I hope it is finally done. 
Ready.

**Comment by Firestorm01X2** [comment]
Don't miss this please:
* **All Leadwork guns can't install attachment now except of accesories. Now it applies to L523 modular system and separate guns including guns for civilian use. Instead guns already comes with wide array of attachments.** 

It can be easily reverted for civilan guns, hovewer. But I want to make Leadworks "modular" manufacturer.

**Pull request merged**